### PR TITLE
[FLINK-28118] Introduce TableStoreOptions to merge all options

### PR DESCRIPTION
**Pull request by hililiwei**
Currently we have OPTIONS in various places, which is harder to maintain, we could have just one class that contains all the OPTIONS, including

MergeTreeOptions
FileStoreOptions
LogOptions

### CODE REVIEW

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/TableStoreOptions.java [hunk 1]
```diff
@@ -0,0 +1,583 @@
...
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static org.apache.flink.configuration.ConfigOptions.key;
+import static org.apache.flink.configuration.description.TextElement.text;
+import static org.apache.flink.table.store.utils.OptionsUtils.formatEnumOption;
+
+/** Options for table store. */
+public class TableStoreOptions implements Serializable {
```
Comment (JingsongLi) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/TableStoreOptions.java:45]: Just `Options`?

File: flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/AbstractTableStoreFactory.java [hunk 2]
```diff
@@ -139,7 +138,13 @@ static DynamicTableFactory.Context createLogContext(DynamicTableFactory.Context
 
     static DynamicTableFactory.Context createLogContext(
             DynamicTableFactory.Context context, Map<String, String> options) {
-        return new TableStoreDynamicContext(context, filterLogStoreOptions(options));
+        return new FactoryUtil.DefaultDynamicTableContext(
```
Comment (JingsongLi) [inline flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/AbstractTableStoreFactory.java:141]: revert this?

**Comment by JingsongLi** [comment]
Thanks @hililiwei for your contribution. left two comments.

**Review by JingsongLi — APPROVED** [review]
Looks good to me! Thanks @hililiwei

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/TableStoreOptions.java (hunk 1 above)
Comment (JingsongLi) [reply flink-table-store-core/src/main/java/org/apache/flink/table/store/TableStoreOptions.java]: Just rename, I will append a commit.

**Comment by hililiwei** [comment]
Thank you for your help with the modification.

**Pull request merged**